### DISCUSSION TITLE
What exactly does 'install' command do?

### DISCUSSION THREAD

**Discussion by 097115** [aclap-dev/video-downloadhelper#9] [Help and Q&A] [answered]
### Your question

Hi! 

`vdhcoapp` help mentions `install` command and describes it like that:

 > register browser JSON files in browser-specific locations.

But what is the process exactly? Does it somehow inform browsers about the data stored in JSON available via `--info` option? 

And can I do this installation manually, like maybe manually copying the said JSON to some browser specific location?

I'm particularly interested in Firefox on Linux -- I see [this](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Native_manifests) mentioned in the README but would be grateful for some more details (especially about the exact paths).

Thanks :)

### Addon & CoApp version and details

`vdhcoapp` version: 2.0.13
`firefox` version 121.0.1

**Comment by paulrouget** [comment]
The install command creates JSON files into these directories: https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Native_manifests#linux - it also runs some flatpak magic commands to make it so that the coapp is seen by sandboxed browsers.

Then Firefox search for JSON files in these directories, and it runs the command and communicate with the binary through regular pipes (stdin/stdout).

You can look at the code yourself right here: https://github.com/aclap-dev/vdhcoapp/blob/master/app/src/native-autoinstall.js

**Reply by 097115** [reply]
Thanks, @paulrouget !

So, basically, something like this:

    vdhcoapp --info > ~/.mozilla/native-messaging-hosts/net.downloadhelper.coapp.json

should work? As I understand, Firefox looks for the configuration JSON in one of the mentioned directories, and the filename is the package id + `.json`, is this correct?

Because after doing this, Firefox still doesn't think the companion app is installed :)

(Just in case, my Firefox is not flatpaked but just a usual binary .tag.gz install.)

**Reply by paulrouget** [reply, marked as answer]
> vdhcoapp --info > ~/.mozilla/native-messaging-hosts/net.downloadhelper.coapp.json

No. That output is unrelated to net.downloadhelper.coapp.json.

But why don't you run the `vdhcoapp install` command to just create the file?

The json format should be:

```json
{
  "type": "stdio",
  "allowed_extensions": [
    "weh-native-test@downloadhelper.net",
    "video-downloadhelper-dev@aclap.com",
    "{b9db16a4-6edc-47ec-a1f4-b86292ed211d}"
  ],
  "name": "net.downloadhelper.coapp",
  "description": "Video DownloadHelper companion app",
  "path": "<path to vdhcoapp binary>"
}

**Reply by 097115** [reply]
Thanks again, Paul!
